knownvalue/string_func.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package knownvalue
+
+import "fmt"
+
+var _ Check = stringFunc{}
+
+type stringFunc struct {
+	checkFunc func(v string) error
+}
+
+// CheckValue determines whether the passed value is of type string, and
+// returns no error from the provided check function
+func (v stringFunc) CheckValue(value any) error {
```

Comment:
Can this be a generic function? I'm not super versed in Go's generic functionality, so I'm not sure if you can run a type assertion with a type parameter.

Reply:
While it’s definitely possible to write a generic function for this, which would involve some type assertions and reflection, I personally prefer our approach of offering typed functions like [knownvalue.Int32Exact](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-testing/knownvalue#Int32Exact), [knownvalue.StringRegexp](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-testing/knownvalue#StringRegexp), etc., in our libraries. I believe that a `Int32Func`, for example, aligns better with the current API design, as it’s more intuitive for users who are already accustomed to working with these typed functions.

It’s an option to go the generic route, but I think sticking with typed functions here makes it easier for users to follow. Does that make sense?